src/ui/components/pages/swap/hooks/useSwappableAssetTickersByVendor.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import { SwapVendor } from 'swap/constants';
+import { useAppSelector } from 'ui/store';
+import { assetIds } from 'assets/constants';
+import { NetworkName } from 'networks/types';
+
+export type SwappableAssetIdRecord = Record<NetworkName.Mainnet, string[]>;
+
+export const useSwappableAssetTickersByVendor = (): {
+  swappableAssetTickersByVendor:
+    | Record<SwapVendor, Set<string>>
+    | Record<string, unknown>;
+  swappableAssetTickers: string[];
+  swappableAssetIds:
+    | Record<NetworkName.Mainnet, string[]>
+    | Record<string, unknown>;
+} => {
+  const data = useAppSelector(state => state.swappableAssetsFromVendor);
+
+  const swappableAssetTickers = useMemo(
+    () =>
+      Array.from(
+        new Set(
+          Object.values(data).flatMap(tickersSet =>
+            tickersSet instanceof Set ? Array.from(tickersSet) : []
+          )
+        )
+      ),
+    [data]
+  );
```

Comment:
This seems wrong. `data` would be `Record<SwapVendor, AssetId>`, not ticker.

So `swappableAssetTickers` actually will contain `swappableAssetIds`.

Reply:
You can also get asset tickers by asset ID from the state, see `state.assetTickers`.

Reply:
Fixing